packages/frontend/src/components/Dialog/Dialog.tsx

```diff
@@ -27,25 +27,74 @@ const Dialog = React.memo<Props>(
     canEscapeKeyClose = true,
     width = DEFAULT_WIDTH,
     height,
+    fullscreenTransparent = false,
+    fixed,
     ...props
   }) => {
+    const dialog = useRef<HTMLDialogElement>(null)
+
+    const onClick = canOutsideClickClose
+      ? (ev: React.MouseEvent<HTMLDialogElement>) => {
+          if (!dialog.current) {
+            return
+          }
+          ev.stopPropagation()
+          // that is the way to check if we clicked on dialog::backdrop
+          const rect = dialog.current.getBoundingClientRect()
+          const isInDialog =
+            rect.top <= ev.clientY &&
+            ev.clientY <= rect.top + rect.height &&
+            rect.left <= ev.clientX &&
+            ev.clientX <= rect.left + rect.width
+          if (!isInDialog) {
+            const cancelEvent = new Event('cancel')
+            dialog.current.dispatchEvent(cancelEvent)
+            // cancel event doesn't trigger dialog close
+            dialog.current.close()
+          }
```

Comment:
> If you open settings, then Tab-navigate to "Chats and Media" and press spacebar, the settings dialog will simply close.

This is happening because of this code. When you use `spacebar` to click, `clientY` and `clientX` would be unset, and it will close the dialog.